rclpy/threading/data_plotter.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2016 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from copy import deepcopy
+from threading import Lock, Thread
+import time
+
+import matplotlib.pyplot as plt
+import rclpy
+
+from std_msgs.msg import Int64
+
+
+"""
+This script starts a node with a subscription to a relatively high-frequency topic.
+The data received from the subscription callback calls is plotted in the main thread,
+while the callback calls themselves are processed in a second thread.
+Plot updates are scheduled at a regular interval independent of the rate at which data is received.
+Slow plotting calls will not block the reception of frequent data messages.
+"""
+
+time_between_plot_updates = 1  # time in seconds between plot updates
+
+
+class DataPlotter:
+    """Class for managing the storage and display of data received."""
+
+    def __init__(self):
+        self.received_data = []
+        # As the received data can be accessed by both the main plotting thread and the thread
+        # processing data callbacks, its access must be protected by a lock
+        self.received_data_lock = Lock()
+
+    def data_callback(self, msg):
+        print('Data received: {0}'.format(msg.data))
+
+        # Store received data
+        self.received_data_lock.acquire()
+        self.received_data.append(msg.data)
+        self.received_data_lock.release()
```

Comment:
thank you for this and other tips! I made the changes in https://github.com/ros2/examples/pull/144/commits/71db410c064a14d8e47ae14e338fa0459ffb90bc